File: space-missions/src/api/Models/Mission.cs
using CsvHelper.Configuration.Attributes;

namespace SpaceMissions.Api.Models;

public class Mission
{
    [Name("Company")]
    public string Company { get; set; } = "";

    [Name("Location")]
    public string Location { get; set; } = "";

    [Name("Date")]
    public string Date { get; set; } = "";

    [Name("Time")]
    public string? Time { get; set; }

    [Name("Rocket")]
    public string Rocket { get; set; } = "";

    [Name("Mission")]
    public string Name { get; set; } = "";

    [Name("RocketStatus")]
    public string RocketStatus { get; set; } = "";

    [Name("Price")]
    public string? Price { get; set; }

    [Name("MissionStatus")]
    public string MissionStatus { get; set; } = "";
}

File: space-missions/src/client/src/types/mission.ts
export interface Mission {
  company: string;
  location: string;
  date: string;
  time: string | null;
  rocket: string;
  name: string;
  rocketStatus: string;
  price: string | null;
  missionStatus: string;
}

export interface TopCompany {
  company: string;
  count: number;
}

export interface MissionStatusCounts {
  Success: number;
  Failure: number;
  "Partial Failure": number;
  "Prelaunch Failure": number;
}

export interface Filters {
  company: string;
  missionStatus: string;
  rocketStatus: string;
  startDate: string;
  endDate: string;
}
